imap_processing/idex/IDEXPacketParser.py

```diff
@@ -0,0 +1,120 @@
+                scitype = pkt.data['IDX__SCI0TYPE'].raw_value
+                if scitype == 1:
+                    evtnum += 1
+                    self.epochs.append(pkt.data['SHCOARSE'].derived_value + 20*(10**(-6))*pkt.data['SHFINE'].derived_value)  # Use this as the CDF epoch
+                    self._log_packet_info(evtnum, pkt) # These are for our logs, don't use
+
+                if scitype in [2, 4, 8, 16, 32, 64]:
+                    if scitype not in self.data:
+                        self.data.update({scitype : {}})
+                    if evtnum not in self.data[scitype]:
+                        self.data[scitype][evtnum] = pkt.data['IDX__SCI0RAW'].raw_value
+                    else:
+                        self.data[scitype][evtnum] += pkt.data['IDX__SCI0RAW'].raw_value
+
+        # Parse the waveforms according to the scitype present (high gain and low gain channels encode waveform data differently).
+        self.scitype_to_names = {2: "TOF_High", 4: "TOF_Low", 8: "TOF_Mid", 16: "Target_Low",
+                                32: "Target_High", 64: "Ion_Grid"}
+        datastore = {}
+        self.time_low_sr = []
+        self.time_high_sr = []
+        for scitype in self.data:
+            datastore[self.scitype_to_names[scitype]] = []
+            for evt in self.data[scitype]:
+                datastore[self.scitype_to_names[scitype]].append(self._parse_waveform_data(self.data[scitype][evt], scitype))
+                if self.scitype_to_names[scitype] == 'Target_Low':
+                    self.time_low_sr.append(np.linspace(0, len(datastore['Target_Low'][0]), len(datastore['Target_Low'][0])))
+                if self.scitype_to_names[scitype] == 'TOF_Low':
+                    self.time_high_sr.append(np.linspace(0, len(datastore['TOF_Low'][0]), len(datastore['TOF_Low'][0])))
+
+        self.data = datastore
+        self.numevents = evtnum
+    
+    def _log_packet_info(self, evtnum, pkt):
+        print(f"^*****Event header {evtnum}******^")
+        # Extract the 17-22-bit integer (usually 8)
+        self.lspretrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value >> 16) &  0b1111
+        # Extract the next 4-bit integer (usually 8)
+        self.lsposttrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value >> 12) & 0b1111
+        # Extract the next 6 bits integer (usually 32)
+        self.hspretrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value >> 6) & 0b111111
+        # Extract the first 6 bits (usually 32)
+        self.hsposttrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value) & 0b111111
+        print("HS pre trig sampling blocks: ", self.hspretrigblocks)
+        print("LS pre trig sampling blocks: ", self.lspretrigblocks)
+        print("HS post trig sampling blocks: ", self.hsposttrigblocks)
+        print("LS post trig sampling blocks: ", self.lsposttrigblocks)
+        self.TOFdelay = pkt.data['IDX__TXHDRSAMPDELAY'].raw_value >> 2  # First two bits are padding
```

Comment:
```suggestion
        self.tof_delay = pkt.data['IDX__TXHDRSAMPDELAY'].raw_value >> 2  # First two bits are padding
```
We should try and stick to python variable naming conventions when possible.